### ISSUE TITLE
Visible fake-gen ores on the surface and in caves on Chunk boundaries

### ISSUE THREAD

**Issue by ProgrammerDan** [lishid/Orebfuscator#157]
As reported here: https://www.spigotmc.org/threads/orebfuscator.145252/page-10#post-2394534

and explained / illustrated here: https://www.spigotmc.org/threads/orebfuscator.145252/page-10#post-2395414

**Comment by Aleksey-Terzi** [comment]
I know about this problem and even tried to solve it, but this ended not very good when we tested it for 50+ players.
Last time I tried to reload chunks once their neighbors were loaded by server, but my implementation was very performance expensive.
I think you remember we had a performance problem for Orebfuscator on Dev 3.0 lunch.

Ok, I will start thinking about the problem and how better to solve it :)

**Comment by ProgrammerDan** [comment]
RIGHT, I forgot about that. Thanks, feel free to PM me / bounce ideas

**Comment by jacob1** [comment]
I've been noticing this a lot on my server too, and players have been complaining about it. At first I thought it was random, but after seeing this I checked the places it appears in, and they were all chunk boundaries.

I use mode 2 so that at least players can tell when orbfuscator is lying to them, but maybe I should use mode 1 so they just see stone and don't complain ...

**Comment by CloudeLecaw** [comment]
i agree with jacob1, i also had noticed this problem on my server with mode 2, also players complained about it. And i also thing about using the mode 1 maybe so the player just see a stone instead of fake orbs :3

**Comment by melinstagibson** [comment]
i did what cloudleclaw did aswell. it turns out tho that it obfuscated visible ores as stone now aswell thus making them invisible :/ i didnt remember having that many issues with fake blocks tho, it worked better for me on 4.3.0 than 4.3.3

**Comment by RoboMWM** [comment]
As a mitigation, you could try only obfuscating blocks that are below a certain y coordinate (this is the case for paper's antixray, which is configurable, though it's based on chunk sections instead of exact y coordinate).

**Comment by Aleksey-Terzi** [comment]
I'm supposing this message was addresses to me.

Thank for suggestion, but I don't think this really will solve problem (but at the same time perhaps would be good to have option), because of most of walls with "fake blocks" are in the caves - we we should do obfuscation.

Problem happens because of problem how to obfuscate edge blocks when adjacent chunk is not loaded.
I'm doing assumption that adjacent block form "non-loaded" chunk is solid and therefore if edge blocks from other sides is covered by solid blocks - it will be obfuscated.

If render distance on server is large - this issue almost never happen, and it will happen more often with smaller render distance.
My last try to solve it by re-sending chunks when adjacent chunks loaded - failed, because of it lowered TPS for server with large amount of players.

**Comment by Raphy123** [comment]
Same problem..

And good performance :(

**Comment by Aleksey-Terzi** [comment]
I think issue is fixed now.
Either wait for new release or recompile it by yourself from this fork: https://github.com/DevotedMC/Orebfuscator/

**Comment by Raphy123** [comment]
Thanks !When did you think the next update ?

**Comment by Aleksey-Terzi** [comment]
I'm not sure, but if you wish you can use my compilation: https://www.dropbox.com/s/tho7si3qg3fmr11/orebfuscator-4.3.4-SNAPSHOT.jar?dl=0  
    
Would be nice to hear if it solved issue for you.

**Comment by ProgrammerDan** [comment]
Should be in a position to release this weekend at the latest, maybe sooner

**Comment by alexander-q** [comment]
Hi Aleksey-Terzi. I tried the provided snapshot, but thereafter encountered hard crashes every few hours.

**Comment by ForceUpdate1** [comment]
@Aleksey-Terzi I have used your fork, but it sill not working
https://prnt.sc/iaksab

**Comment by DArkes1312** [comment]
Any news ? My players are going insane.

**Comment by RoboMWM** [comment]
> My players are going insane.

Use mode 1 for now then.
